File: tests-cli/features/step_definitions/steps.rb
When('the hello world function is called') do
  run_command_and_stop('hello', fail_on_error: true)
  @output = last_command_started.stdout.chomp
end

When('I greet {string}') do |name|
  run_command_and_stop("hello #{name}", fail_on_error: true)
  @output = last_command_started.stdout.chomp
end

Then('the output is {string}') do |expected|
  expect(@output).to eq(expected)
end

File: tests-js/features/step_definitions/steps.js
const { Given, When, Then } = require('@cucumber/cucumber');
const assert = require('assert');
const hello = require('../../../bin/hello.js');

let result;
let input;

When('I greet {string}', function (name) {
  result = hello.greet(name);
});

Then('the output is {string}', function (expected) {
  assert.strictEqual(result, expected);
});

Given('a blank string', function () {
  input = '';
});

When('we call parse', function () {
  result = JSON.parse(hello.parse(input));
});

Then('it returns an empty json object', function () {
  assert.deepStrictEqual(result, {});
});
